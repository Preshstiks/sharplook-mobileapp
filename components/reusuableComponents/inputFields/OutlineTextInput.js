import React, { useState } from "react";
import { TextInput, StyleSheet, View, Text } from "react-native";

export default function OutlineTextInput({
  value,
  onChangeText,
  placeholder,
  secureTextEntry = false,
  keyboardType = "default",
  style,
  label,
  error,
  touched,
  editable = true,
  ...props
}) {
  const [isFocused, setIsFocused] = useState(false);
  // Filter out key from props to avoid React warning
  const { key, ...rest } = props;
  return (
    <View className="mb-3">
      {label && (
        <Text
          style={{ fontFamily: "poppinsMedium" }}
          className="text-[12px] pb-1"
        >
          {label}
        </Text>
      )}
      <TextInput
        value={value}
        onChangeText={onChangeText}
        placeholder={placeholder}
        placeholderTextColor="#C4C4C4"
        secureTextEntry={secureTextEntry}
        keyboardType={keyboardType}
        editable={editable}
        style={[
          { fontFamily: "poppinsRegular", textAlignVertical: "top" },
          style,
        ]}
        className={`border px-4 py-4 text-[12px] rounded-[12px] ${isFocused ? "border-primary" : error && touched ? "border-[#FF0000]" : "border-[#F9BCDC]"}`}
        onFocus={() => setIsFocused(true)}
        onBlur={() => setIsFocused(false)}
        {...rest}
      />
      {error && touched && <Text style={styles.errorText}>{error}</Text>}
    </View>
  );
}

export function OutlineTextAreaInput({
  value,
  onChangeText,
  placeholder,
  style,
  label,
  numberOfLines = 4,
  minHeight = 100,
  error,
  touched,
  ...props
}) {
  const [isFocused, setIsFocused] = useState(false);
  // Filter out key from props to avoid React warning
  const { key, ...rest } = props;
  return (
    <View className="mb-3">
      {label && (
        <Text
          style={{ fontFamily: "poppinsMedium" }}
          className="text-[12px] pb-1"
        >
          {label}
        </Text>
      )}
      <TextInput
        value={value}
        onChangeText={onChangeText}
        placeholder={placeholder}
        placeholderTextColor="#C4C4C4"
        multiline
        numberOfLines={numberOfLines}
        className={`border px-4 text-[12px] py-4 rounded-[12px] ${isFocused ? "border-primary" : error && touched ? "border-[#FF0000]" : "border-[#F9BCDC]"}`}
        style={[
          { fontFamily: "poppinsRegular", minHeight, textAlignVertical: "top" },
          style,
        ]}
        onFocus={() => setIsFocused(true)}
        onBlur={() => setIsFocused(false)}
        {...rest}
      />
      {error && touched && <Text style={styles.errorText}>{error}</Text>}
    </View>
  );
}
const styles = StyleSheet.create({
  inputContainer: {
    borderWidth: 1,
    borderColor: "transparent",
    borderRadius: 8,
    paddingHorizontal: 12,
    paddingVertical: 2,
    marginBottom: 12,
    backgroundColor: "#fff",
  },
  input: {
    height: 44,
    fontSize: 15,
    color: "#201E1F",
    fontFamily: "poppinsRegular",
  },
  label: {
    fontFamily: "poppinsRegular",
    fontSize: 13,
    color: "#201E1F",
    marginBottom: 2,
    marginTop: 6,
  },
  errorText: {
    color: "#FF0000",
    fontSize: 12,
    fontFamily: "poppinsRegular",
    marginTop: 4,
    marginLeft: 12,
  },
});
