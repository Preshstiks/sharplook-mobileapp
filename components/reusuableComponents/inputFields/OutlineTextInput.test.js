import React from "react";
import { render, fireEvent } from "@testing-library/react-native";
import OutlineTextInput, { OutlineTextAreaInput } from "./OutlineTextInput";

describe("OutlineTextInput", () => {
  const mockOnChangeText = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("renders label and placeholder", () => {
    const { getByText, getByPlaceholderText } = render(
      <OutlineTextInput
        label="Username"
        placeholder="Enter your username"
        value=""
        onChangeText={mockOnChangeText}
      />
    );

    expect(getByText("Username")).toBeTruthy();
    expect(getByPlaceholderText("Enter your username")).toBeTruthy();
  });

  it("calls onChangeText when typing", () => {
    const { getByPlaceholderText } = render(
      <OutlineTextInput
        placeholder="Type here"
        value=""
        onChangeText={mockOnChangeText}
      />
    );

    fireEvent.changeText(getByPlaceholderText("Type here"), "Hello");
    expect(mockOnChangeText).toHaveBeenCalledWith("Hello");
  });

  it("changes border color on focus", () => {
    const { getByPlaceholderText, toJSON } = render(
      <OutlineTextInput
        placeholder="Focus me"
        value=""
        onChangeText={() => {}}
      />
    );

    const input = getByPlaceholderText("Focus me");
    fireEvent(input, "focus");

    expect(toJSON()).toMatchSnapshot(); // Snapshot will capture focused border class
  });

  it("shows error when error and touched are true", () => {
    const { getByText } = render(
      <OutlineTextInput
        placeholder="Email"
        value=""
        onChangeText={() => {}}
        error="Required field"
        touched={true}
      />
    );

    expect(getByText("Required field")).toBeTruthy();
  });

  it("does not show error when touched is false", () => {
    const { queryByText } = render(
      <OutlineTextInput
        placeholder="Email"
        value=""
        onChangeText={() => {}}
        error="Required field"
        touched={false}
      />
    );

    expect(queryByText("Required field")).toBeNull();
  });

  it("is not editable when editable is false", () => {
    const { getByPlaceholderText } = render(
      <OutlineTextInput
        placeholder="Disabled"
        value=""
        onChangeText={() => {}}
        editable={false}
      />
    );

    const input = getByPlaceholderText("Disabled");
    expect(input.props.editable).toBe(false);
  });
});

describe("OutlineTextAreaInput", () => {
  const mockOnChangeText = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("renders multiline textarea with label", () => {
    const { getByText, getByPlaceholderText } = render(
      <OutlineTextAreaInput
        label="Description"
        placeholder="Enter description"
        value=""
        onChangeText={mockOnChangeText}
      />
    );

    expect(getByText("Description")).toBeTruthy();
    expect(getByPlaceholderText("Enter description")).toBeTruthy();
  });

  it("calls onChangeText when text changes", () => {
    const { getByPlaceholderText } = render(
      <OutlineTextAreaInput
        placeholder="Write here"
        value=""
        onChangeText={mockOnChangeText}
      />
    );

    fireEvent.changeText(getByPlaceholderText("Write here"), "Some text");
    expect(mockOnChangeText).toHaveBeenCalledWith("Some text");
  });

  it("shows error when error and touched are true", () => {
    const { getByText } = render(
      <OutlineTextAreaInput
        placeholder="Notes"
        value=""
        onChangeText={() => {}}
        error="Field required"
        touched={true}
      />
    );

    expect(getByText("Field required")).toBeTruthy();
  });
});
