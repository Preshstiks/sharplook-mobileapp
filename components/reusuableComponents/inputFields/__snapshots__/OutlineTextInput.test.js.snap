// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OutlineTextInput changes border color on focus 1`] = `
<View
  className="mb-3"
>
  <TextInput
    className="border px-4 py-4 text-[12px] rounded-[12px] border-primary"
    editable={true}
    keyboardType="default"
    onBlur={[Function]}
    onChangeText={[Function]}
    onFocus={[Function]}
    placeholder="Focus me"
    placeholderTextColor="#C4C4C4"
    secureTextEntry={false}
    style={
      [
        {
          "fontFamily": "poppinsRegular",
          "textAlignVertical": "top",
        },
        undefined,
      ]
    }
    value=""
  />
</View>
`;
